srv/salt/ceph/monitoring/prometheus/update_service_discovery.sls

```diff
@@ -21,8 +21,10 @@
     - dir_mode: 755
     - clean: True
     - makedirs: True
+{% if salt.saltutil.runner('select.minions', cluster='ceph', host=False) %}
```

Comment:
Just out of curiosity: did you actually run into this? This should only happen when there are no minions or what am I missing?

Reply:
Yes, I run into this problem, but it only happens when running the CLI. The CLI parses all state files that are part of a stage. The parsing requires that all jinja statements are evaluated so that we can actually know what is going to execute. This jinja statement assumes that `select.minions` will return something because the configuration was already applied in the beginning of the stage, but the problem is that we are just parsing and not executing the state files, and therefore the configuration was not applied yet.